src/ResourceManager/Common/Commands.Common.Strategies/Extensions.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Azure.Commands.Common.Strategies
+{
+    public static class Extensions
+    {
+        public static IEnumerable<T> EmptyIfNull<T>(this IEnumerable<T> value)
+            => value ?? Enumerable.Empty<T>();
+
+        public static V GetOrNull<K, V>(this IDictionary<K, V> dictionary, K key)
+            where V : class
+        {
+            V result;
+            return dictionary.TryGetValue(key, out result) ? result : null;
```

Comment:
This can be

```csharp
var result = dictionary.TryGetValue(key, out var result);
return result;
```

since `result` will be `default(V)` after the `TryGetValue` call, and you have specified `where V : class`.  The default value of class types is `null`;

Reply:
Sadly, but we use Visual Studio 2015, C# 6.